llvm/lib/IR/Function.cpp

```diff
@@ -963,8 +963,9 @@ bool Function::isTargetIntrinsic() const {
 /// Find the segment of \c IntrinsicNameTable for intrinsics with the same
 /// target as \c Name, or the generic table if \c Name is not target specific.
 ///
-/// Returns the relevant slice of \c IntrinsicNameTable
-static ArrayRef<const char *> findTargetSubtable(StringRef Name) {
+/// Returns the relevant slice of \c IntrinsicNameTable and the target name.
+static std::pair<ArrayRef<const char *>, StringRef>
+findTargetSubtable(StringRef Name) {
```

Comment:
As a side note, this function seems to be only called from `lookupIntrinsicID` and is small enough to be inlined. May be we can do that as a separate NFC?

Reply:
It's not exactly a small function, let the inliner do it

Reply:
Fair enough, we can leave it as is.